src/collection/AbstractList.php

```diff
@@ -38,4 +46,63 @@ abstract class AbstractList extends AbstractCollection {
 	public function reverseSort($cmp = null): self {
 		return $this->sort($cmp)->reverse();
 	}
+
+	/**
+	 * Returns the index of the given element or null if the element can't be found
+	 *
+	 * @param mixed $element
+	 *
+	 * @return int the index for the given element
+	 */
+	public function indexOf($element): ?int {
+		$index = $this->parentIndexOf($element);
+
+		return null === $index ? $index : (int) $index;
+	}
+
+	/**
+	 * Searches the array with a given callback and returns the index for the last element if found.
+	 *
+	 * The callback function takes one or two parameters:
+	 *
+	 *     function ($element [, $query]) {}
+	 *
+	 * The callback must return a boolean
+	 * When it's passed, $query must be the first argument:
+	 *
+	 *     - find($query, callback)
+	 *     - find(callback)
+	 *
+	 * @param array $arguments
+	 *
+	 * @return int|null the index or null if it hasn't been found
+	 */
+	public function findLastIndex(...$arguments): ?int {
+		$lastIndex = $this->parentFindLastIndex(...$arguments);
+
+		return null === $lastIndex ? $lastIndex : (int) $lastIndex;
+	}
+
+	/**
+	 * Searches the array with a given callback and returns the index for the first element if found.
+	 *
+	 * The callback function takes one or two parameters:
+	 *
+	 *     function ($element [, $query]) {}
+	 *
+	 * The callback must return a boolean
+	 * When it's passed, $query must be the first argument:
+	 *
+	 *     - find($query, callback)
+	 *     - find(callback)
+	 *
+	 * @param array $arguments
+	 *
+	 * @return int|null the index or null if it hasn't been found
+	 */
+	public function findIndex(...$arguments): ?int {
+		$index = $this->parentFindIndex(...$arguments);
+
+		return null === $index ? $index : (int) $index;
```

Comment:
Please no yoda style... hard to read it is.

Reply:
fixed